go/engine/saltpack_encrypt.go

```diff
@@ -199,116 +179,11 @@ func (e *SaltpackEncrypt) Run(m libkb.MetaContext) (err error) {
 		Sender:             senderDH,
 		SenderSigning:      senderSigning,
 		Binary:             e.arg.Opts.Binary,
-		EncryptionOnlyMode: e.arg.Opts.EncryptionOnlyMode,
+		EncryptionOnlyMode: encryptionOnlyMode,
 		SymmetricReceivers: symmetricReceivers,
 		SaltpackVersion:    saltpackVersion,
 
 		VisibleRecipientsForTesting: e.visibleRecipientsForTesting,
 	}
 	return libkb.SaltpackEncrypt(m.G(), &encarg)
 }
-
-func (e *SaltpackEncrypt) getCryptKeys(m libkb.MetaContext, name string) (keybase1.GetTLFCryptKeysRes, error) {
-	xp := m.G().ConnectionManager.LookupByClientType(keybase1.ClientType_KBFS)
-	if xp == nil {
-		return keybase1.GetTLFCryptKeysRes{}, libkb.KBFSNotRunningError{}
-	}
-	cli := &keybase1.TlfKeysClient{
-		Cli: rpc.NewClient(xp, libkb.NewContextifiedErrorUnwrapper(m.G()), libkb.LogTagsFromContext),
-	}
-	return cli.GetTLFCryptKeys(m.Ctx(), keybase1.TLFQuery{
-		TlfName:          name,
-		IdentifyBehavior: keybase1.TLFIdentifyBehavior_SALTPACK,
-	})
-}
-
-func (e SaltpackEncrypt) completeAndCanonicalize(m libkb.MetaContext, tlfName string) (keybase1.CanonicalTLFNameAndIDWithBreaks, error) {
-	username := m.G().Env.GetUsername()
-	if len(username) == 0 {
-		return keybase1.CanonicalTLFNameAndIDWithBreaks{}, libkb.LoginRequiredError{}
-	}
-
-	// Prepend username in case it's not present. We don't need to check if it
-	// exists already since CryptKeys calls below transforms the TLF name into a
-	// canonical one.
-	//
-	// This makes username a writer on this TLF, which might be unexpected.
-	// TODO: We should think about how to handle read-only TLFs.
-	tlfName = string(username) + "," + tlfName
-
-	resp, err := e.getCryptKeys(m, tlfName)
-	if err != nil {
-		return keybase1.CanonicalTLFNameAndIDWithBreaks{}, err
-	}
-
-	return resp.NameIDBreaks, nil
-}
-
-// TODO: Make sure messages that encrypt only to self are working properly.
-func (e *SaltpackEncrypt) makeSymmetricReceivers(m libkb.MetaContext) ([]saltpack.ReceiverSymmetricKey, error) {
```

Comment:
is this removed or moved?

Reply:
This is now moved to a dedicated engine (for testing only) in saltpackkeys.